library/psa_crypto.c

```diff
@@ -5197,11 +5204,34 @@ static psa_status_t psa_tls12_prf_set_seed( psa_tls12_prf_key_derivation_t *prf,
     return( PSA_SUCCESS );
 }
 
+static psa_status_t psa_tls12_prf_set_salt( psa_tls12_prf_key_derivation_t *prf,
+                                            const uint8_t *data,
+                                            size_t data_length )
+{
+    if( prf->state != PSA_TLS12_PRF_STATE_SEED_SET )
+        return( PSA_ERROR_BAD_STATE );
+
+    if( data_length != 0 )
+    {
+        prf->salt = mbedtls_calloc( 1, data_length );
```

Comment:
On second though, why do we use a dynamic buffer here, why we use a statically-size one for the PMS, which is larger? This one is part of the `psa_tls12_prf_key_derivation_t` while the PMS buffer is local to a function, but does it justify the difference? (This is a question, not necessarily asking for a change.)

Reply:
It's done in this way because of consistency with the other inputs (seed, secret, etc.). For PMS static buffer was used. 

From my perspective if there is a problem with the optimal size of the static buffer for `PMS` it should be allocated dynamically.

Reply:
> From my perspective if there is a problem with the optimal size of the static buffer for PMS it should be allocated dynamically.

Yes, I tend to agree. Since we're already doing dynamic allocation as part of the operation anyway, let's keep advantage of it and avoid the thorny sizing issues associated with a static buffer.